src/js/actions/authActions.js

```diff
@@ -0,0 +1,85 @@
+import fetch from 'fetch-everywhere';
+import {push} from 'react-router-redux';
+import urlFor from '../helpers/urlFor';
+
+const requestLogout = () => ({
+  type: 'LOGOUT_REQUEST',
+  isFetching: true,
+  isAuthenticated: true
+});
+
+const receiveLogout = () => ({
+  type: 'LOGOUT_SUCCESS',
+  isFetching: false,
+  isAuthenticated: false
+});
+
+// Logs the user out
+export const logoutUser = () => (dispatch) => {
+  dispatch(requestLogout());
+  localStorage.removeItem('idToken');
+  localStorage.removeItem('store');
+  dispatch(receiveLogout());
+  dispatch(push('/'));
+};
+
+const requestLogin = (code) => ({
+  type: 'LOGIN_REQUEST',
+  isFetching: true,
+  isAuthenticated: false,
+  code
+});
+
+const receiveLogin = (token) => ({
+  type: 'LOGIN_SUCCESS',
+  isFetching: false,
+  isAuthenticated: true,
+  user: token
+});
+
+const loginError = (err) => ({
+  type: 'LOGIN_FAILURE',
+  isFetching: false,
+  isAuthenticated: false,
+  err
+});
+
+export const loginUser = (code) => (dispatch) => {
+  // API Route for Slack Authentication
+  const url = urlFor('/api/v1/auth/slack');
```

Comment:
Works for me - If you have a reason to be using them then great. There are just a lot of constants here, which I rarely see in the wild outside of declarations for things like PI. 

Naming convention does dictate the use of ALL_CAP names for constants though. You will see this in any language for the most part; it's one of the few things that everyone seems to agree on.